### DISCUSSION TITLE
Joint Force/Torque Sensing?

### DISCUSSION THREAD

**Discussion by EChoudhury** [haosulab/SAPIEN#156] [Q&A] [answered]
Hi all, 

Is there a way to calculate the force/torque values of joints, similar to a force/torque sensor? Is this currently supported? 

Thanks for the help!

**Comment by StoneT2000** [comment, marked as answer]
Hi, we will expose an API for fetching force+torque values of joints on articulations in the future. We are currently still testing them. We have some progress on it here already but it is not in the public pypi package yet: https://github.com/haosulab/SAPIEN/commit/3037022b43a7466d037dd61877c72e4b18f8aacc

**Reply by EChoudhury** [reply]
Ah ok, understood, thanks for the info - and for the quick response!

**Reply by Z5cc** [reply]
Is there any progress on this?
Also, will this feature be enabled for simulations on the GPU?

Thx in advance!

**Reply by fbxiang** [reply]
This feature has been added to SAPIEN for a while. And I see it used in ManiSkill here and you can probably use this as reference.

https://github.com/haosulab/ManiSkill/blob/9e0efc348f3821543548eca6d878df5ebe8606e8/mani_skill/utils/structs/articulation.py#L596
